### PR TITLE
Table: Convert examples to use Sandpack

### PR DESCRIPTION
**Pull request by diyorbek**
#### What changed?

Migrated examples in the docs from the old editor to the sandpack editor for the Table component.

#### Why?

Migrated examples in the docs from the old editor to the sandpack editor for the Table component.

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-4873)

### Checklist

- [n/a] Added unit and Flow Tests
- [n/a] Added documentation + accessibility tests
- [n/a] Verified accessibility: keyboard & screen reader interaction
- [x] Checked dark mode, responsiveness, and right-to-left support
- [n/a] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/examples/table/alignContent.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+// @flow strict
+import { type Node } from 'react';
+import { Table, Box, Label, Checkbox, Text } from 'gestalt';
+
+function HeaderRow({ id }: any) {
```
[the remark this answers is not in the document]
Comment (diyorbek) [reply docs/examples/table/alignContent.js:5]: Also I hope that it is OK to `:any` for prop types here. Sandpack is removing the type annotations anyway and I don't think we want to write extra code just to make ESlint happy :) .

**Comment by diyorbek** [comment]
@dangerismycat @AlbertCarreras I didn't add the paddings back to the examples as them seemed more appealing without them. I hope it is OK.

### Before:
Screenshot 2023-05-30 at 20 07 06

### After:
Screenshot 2023-05-30 at 20 06 48

File: docs/examples/table/dontAddSoMuchContent.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+              {title}
+            </Text>
+          </Table.HeaderCell>
+        ))}
+      </Table.Row>
+    </Table.Header>
+  );
+}
+
+// $FlowIgnore
```
Comment (diyorbek) [inline docs/examples/table/dontAddSoMuchContent.js:21]: @AlbertCarreras @dangerismycat Ignored flow type errors in "Don't" examples as the source code is not presented.

File: docs/examples/table/alignContent.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (diyorbek) [reply docs/examples/table/alignContent.js]: @AlbertCarreras @dangerismycat Ignored flow type errors in "Don't" examples as the source code is not presented.

File: docs/examples/table/dontAddSoMuchContent.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+              {title}
+            </Text>
+          </Table.HeaderCell>
+        ))}
+      </Table.Row>
+    </Table.Header>
+  );
+}
+
+// $FlowIgnore[unclear-type]
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/examples/table/dontAddSoMuchContent.js:21]: I'll approve. I'd type those examples, just as a good practice so we don't get in the habit of disabling. When we move to TS that probably will help the migration

**Review by AlbertCarreras — APPROVED** [review]

**Review by AlbertCarreras — APPROVED** [review]

**Pull request merged**